tests/validator/test_validator_api.py

```diff
@@ -12,6 +13,11 @@
 
 
 class TestValidatorApi(TestValidatorBase):
+
+    # jsonschema 4.* is more restrictive than 3, so we need a separate check
+    jsonschemaMajorVersion = int(jsonschema.__version__.split(".")[0])
+    _error_definitionuri = "error_definitionuri" if jsonschemaMajorVersion > 3 else "error_definitionuri_jsonschema3"
```

Comment:
I am not super fan of conditional unit test. Do you know any risk getting rid of jsonschema 3.x?

Reply:
this was a direct respond to the @jfuss's review of #2441, which indicated hesitancy to remove 3.x compatibility.

Reply:
(created this PR in hopes that this would get this across the finish line as it is a bit of a nuisance for anyone developing with/using moto)

Reply:
I am ok with moving forward if it blocks you with a TODO note in requirement

Reply:
added TODO in base.txt